test/setupMeasure_test.js

```diff
@@ -0,0 +1,111 @@
+goog.module('measurementLibrary.measure.testing.setup');
+goog.setTestOnly();
+
+/**
+ * Run a test, once assuming that the measure snippet fired before the call
+ * to setup, then again assuming that the measure snippet fired after the call
+ * to setup.
+ * When the measurement library is added to a html page, it is loaded
+ * asynchronously with a setup snippet. Hence, the two code blocks can
+ * run in either order. This function ensures that the library has the same
+ * behavior no matter which code is executed first.
+ *
+ * @param {function(!Array<*>):undefined} config The configuration
+ *     to be done in the measure code snippet.
+ * @param {function(!Array<*>):undefined} test The tests to run
+ *     after both the code snippet and setupMeasure function have fired.
+ */
+function executeSnippetBeforeAndAfterSetup(config, test) {
+  let snippet;
+  let dataLayer;
+  beforeEach(() => {
+    dataLayer = [];
+    // The code snippet that is run asynchronously.
```

Comment:
Can you add some additional context as to what this is doing? It looks to me like this is just setting up a fake version of "measure". Is this temporary? If so, can you add a "TODO"?

Reply:
This is not temporary, the html file that calls measurement library has two components - the script tag that calls the library and the script tag that defines user settings. The code in this before each is the latter script tag.

Reply:
Sounds good.